service/lib/agama/registration.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "fileutils"
+require "yast"
+require "ostruct"
+require "suse/connect"
+require "y2packager/new_repository_setup"
+
+Yast.import "Arch"
+
+module Agama
+  # Handles everything related to registration of system to SCC, RMT or similar
+  class Registration
+    attr_reader :reg_code
+
+    attr_reader :email
+
+    module Requirement
+      NOT_REQUIRED = :not_required
+      OPTIONAL = :optional
+      MANDATORY = :mandatory
+    end
+
+    # initializes registration with instance of software manager for query about products
+    def initialize(software_manager)
+      @software = software_manager
+    end
+
+    def register(code, email: "")
+      return unless product
+
+      connect_params = {
+        token: code,
+        email: email
+      }
+
+      login, password = SUSE::Connect::YaST.announce_system(connect_params, target_distro)
+      # write the global credentials
+      # TODO: check if we can do it in memory for libzypp
+      SUSE::Connect::YaST.create_credentials_file(login, password)
+
+      # TODO: fill it properly for scc
+      target_product = OpenStruct.new(
+        arch:       Yast::Arch.rpm_arch,
+        identifier: product.id,
+        version:    product.version
+      )
+      activate_params = {}
+      @service = SUSE::Connect::YaST.activate_product(target_product, activate_params, email)
+      Y2Packager::NewRepositorySetup.instance.add_service(@service.name)
```

Comment:
@lslezak I am not sure if we need this call in Agama, at first I thought it adds to libzypp new service, but it looks more that it is just for migration. Am I right?

Reply:
Yes, that class is used to track the new added services or repositories so at the end of migration we can find out which are the new repositories (must be kept in the system) or the old ones (should be removed).

See https://github.com/yast/yast-yast2/blob/master/library/packages/src/lib/y2packager/new_repository_setup.rb#L19-L21

Reply:
note for me: method we will be needed is this one https://github.com/yast/yast-registration/blob/master/src/lib/registration/sw_mgmt.rb#L365 I will try to reduce it as much as possible for currently used features in agama